app/src/main/AndroidManifest.xml

```diff
@@ -14,6 +14,8 @@
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```

Comment:
I don't know that these permissions are necessary, since you are using the application private path in your code, and I didn't need permissions for this.